bciers/apps/registration/app/components/operations/OperationDataGrid.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { useMemo, useState } from "react";
+import DataGrid from "@bciers/components/datagrid/DataGrid";
+import { GridColumnGroupHeaderParams } from "@mui/x-data-grid";
+import { useSession } from "next-auth/react";
+import OperationsActionCell from "@bciers/components/datagrid/cells/OperationsActionCell";
+import HeaderSearchCell from "@bciers/components/datagrid/cells/HeaderSearchCell";
+import { fetchOperationsPageData } from "./Operations";
+import operationColumns from "../datagrid/models/operationColumns";
+import operationGroupColumns from "../datagrid/models/operationGroupColumns";
+import { OperationRow } from "./types";
+
+const OperationSearchCell = ({
+  lastFocusedField,
+  setLastFocusedField,
+}: {
+  lastFocusedField: string | null;
+  setLastFocusedField: (value: string | null) => void;
+}) => {
+  const RenderCell = (params: GridColumnGroupHeaderParams) => {
+    const { groupId, headerName } = params;
+    return (
+      <HeaderSearchCell
+        field={groupId as string}
+        fieldLabel={headerName as string}
```

Comment:
Hmm not sure what the best way fix these type assertions are, do you think something like `groupId ?? ""` is a good idea?

Reply:
Sure